multi_indir test failed due to instruction count mismatch

multi_indir failed due to instruction count mismatch:

https://github.com/DynamoRIO/dynamorio/actions/runs/25870706337/job/76024859154?pr=7898

```
2026-05-14T16:12:47.7469344Z 355: CMake Error at /home/runner/work/dynamorio/dynamorio/suite/tests/runmulti.cmake:72 (message):
2026-05-14T16:12:47.7470334Z 355:   output |Schedule stats tool results:
2026-05-14T16:12:47.7470948Z 355: 
2026-05-14T16:12:47.7471389Z 355:   Total counts:
2026-05-14T16:12:47.7471889Z 355: 
2026-05-14T16:12:47.7472333Z 355:              3 cores
2026-05-14T16:12:47.7474377Z 355:             27 threads: W1.T872894, W1.T872901, W1.T872902, W1.T872904, W1.T872906, W1.T872905, W1.T872884, W1.T872805, W1.T872903, W2.T872905, W2.T872906, W2.T872884, W2.T872904, W2.T872902, W2.T872901, W2.T872894, W2.T872805, W2.T872903, W0.T872906, W0.T872884, W0.T872894, W0.T872904, W0.T872905, W0.T872903, W0.T872805, W0.T872902, W0.T872901
2026-05-14T16:12:47.7476835Z 355:        1839054 instructions
2026-05-14T16:12:47.7519488Z 355:             63 total context switches


2026-05-14T16:12:47.8040610Z 355:   | failed to match expected output |Schedule stats tool results:
2026-05-14T16:12:47.8041159Z 355: 
2026-05-14T16:12:47.8041416Z 355:   Total counts:
2026-05-14T16:12:47.8041705Z 355: 
2026-05-14T16:12:47.8041961Z 355:              3 cores
2026-05-14T16:12:47.8042419Z 355:             27 threads: .*W[012].T872902.*W[012].T872902.*W[012].T872902.*
2026-05-14T16:12:47.8042974Z 355:        1930251 instructions
```

--- a/clients/drcachesim/tests/multi_indir.templatex
+++ b/clients/drcachesim/tests/multi_indir.templatex
@@ -2,7 +2,7 @@ Schedule stats tool results:
 Total counts:
            3 cores
           27 threads: .*W[012].T872902.*W[012].T872902.*W[012].T872902.*
-     1930251 instructions
+     1951728 instructions
 .*
 Core #0 schedule: [A-Za-z_]*
 Core #1 schedule: [A-Za-z_]*

--- a/suite/tests/CMakeLists.txt
+++ b/suite/tests/CMakeLists.txt
@@ -4262,8 +4262,10 @@ if (BUILD_CLIENTS)
         set(tool.switch_file_invariants_rawtemp ON) # no preprocessor
 
         # Test -multi_indir with 3 copies of our sample dir.
+        # We set -sched_exit_if_fraction_inputs_left to 0 so our exact instruction
+        # count check in the template will always match.
         torunonly_simtool(multi_indir ${ci_shared_app}
-          "-multi_indir ${thread_trace_dir}:${thread_trace_dir}:${thread_trace_dir} ${thread_trace_scale} -tool schedule_stats -cores 3"
+          "-multi_indir ${thread_trace_dir}:${thread_trace_dir}:${thread_trace_dir} ${thread_trace_scale} -tool schedule_stats -cores 3 -sched_exit_if_fraction_inputs_left 0"
           "")
         set(tool.multi_indir_rawtemp ON) # no preprocessor
 

